recovery/recovery.go

```diff
@@ -0,0 +1,22 @@
+package recovery
+
+import (
+	"fmt"
+)
+
+// Do is a function wrapper that will recover from a panic and return the error.
```

Comment:
I think this is very nice since it theoretically means our query engine cannot crash a node anymore. But are there maybe some panics we do want to make sure go through? At cockroach we took the approach that all panics from the query engine are treated as errors and that's it. One way to "filter" is to panic with a specific object (e.g. ErrOutOfMemory) that is then checked here.

Reply:
I think at present we don't have a use case to let a panic through so can leave it for now. But we certainly can add future handling here if need be.